File: frontend/types/conversation.ts
export interface MessageResponse {
  id: string;
  role: "user" | "assistant";
  content: string;
  created_at: string;
}

export interface ConversationResponse {
  id: string;
  title: string;
  user_id?: string;
  created_at: string;
  updated_at: string;
  messages: MessageResponse[];
}

export interface ConversationListItem {
  id: string;
  title: string;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/conversation.py
from datetime import datetime
from typing import List, Optional
from uuid import UUID
from pydantic import BaseModel, Field, ConfigDict

class MessageResponseSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    role: str
    content: str
    created_at: datetime

class ConversationResponseSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    user_id: Optional[UUID] = None
    created_at: datetime
    updated_at: datetime
    messages: List[MessageResponseSchema] = []

class ConversationListItemSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    created_at: datetime
    updated_at: datetime

class ConversationRenameRequest(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
